indexing-service/src/main/java/io/druid/indexing/common/task/HadoopTask.java

```diff
@@ -93,7 +93,7 @@ protected ClassLoader buildClassLoader(final TaskToolbox toolbox) throws Excepti
       driverURLs.addAll(Arrays.asList(((URLClassLoader) hadoopLoader).getURLs()));
     }
 
-    final URLClassLoader loader = new URLClassLoader(driverURLs.toArray(new URL[driverURLs.size()]), null);
+    final URLClassLoader loader = new URLClassLoader(driverURLs.toArray(new URL[driverURLs.size()]), Thread.currentThread().getContextClassLoader());
```

Comment:
it doesn't show up otherwise. In fact since I am now going to move UsedSegmentLister setup in HadoopDetermineConfigInnerProcessing.runTask(..) , this change wouldn't be needed.